### PR TITLE
aarch64 platform mimx8mm/p update and new mimx8mn platform

### PR DESCRIPTION
**Pull request by JiafeiPan**
This patchset includes the following updates:
1.  Add new arm64 platform imx8mn platform support.
2. Clean up imx8mm and imx8mp Cortex-A platform support, each platform use two board configuration, one for non-SMP and the other for SMP kernel, both of them can boot from uboot command or Jailhouse inmate cell.
3. Enabled physical UART4 for all board configurations.
4. Update pinctrl_imx driver to be aarch64 compatible.
5. Enabled pinctrl for all these three platforms and it works with UART.

### CODE REVIEW

**Review by carlocaione — COMMENTED** [review]
Nice work. Just a comment about commit messages: please add one even if it a short one.

File: soc/arm64/nxp_imx/imx8m/soc.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/*
```
Comment (gmarull) [inline soc/arm64/nxp_imx/imx8m/soc.h:1]: soc.h is gone for arm64, please remove it.

**Review by gmarull — DISMISSED** [review]

File: soc/arm64/nxp_imx/imx8m/soc.h (hunk 1 above)
Comment (JiafeiPan) [reply soc/arm64/nxp_imx/imx8m/soc.h:1]: Hi, @gmarull , thanks for your comments, but there are still some drivers need soc.h, such as:
drivers/clock_control/clock_control_mcux_ccm.c
drivers/pinctrl/pinctrl_imx.c
drivers/serial/uart_mcux_iuart.c

So how to handle this issue? just create a dumy one? thanks.

Comment (JiafeiPan) [reply soc/arm64/nxp_imx/imx8m/soc.h:1]: Hi, @gmarull , thanks for your comments, but there are still some drivers need soc.h, such as:
drivers/clock_control/clock_control_mcux_ccm.c
drivers/pinctrl/pinctrl_imx.c
drivers/serial/uart_mcux_iuart.c

So how to handle this issue? just create a dumy one? thanks.

**Comment by JiafeiPan** [comment]
> Nice work. Just a comment about commit messages: please add one even if it a short one.

Thanks your comments, will update it.

File: soc/arm64/nxp_imx/imx8m/soc.h (hunk 1 above)
Comment (gmarull) [reply soc/arm64/nxp_imx/imx8m/soc.h:1]: then please fix drivers to include the right headers.

Comment (JiafeiPan) [reply soc/arm64/nxp_imx/imx8m/soc.h]: okay, I removed soc.h from this PR, and created another PR to fix drivers:
https://github.com/zephyrproject-rtos/zephyr/pull/49734

Thanks.

**Comment by JiafeiPan** [comment]
> Nice work. Just a comment about commit messages: please add one even if it a short one.

Done, thanks.

**Review by carlocaione — DISMISSED** [review]

**Review by povergoing — DISMISSED** [review]
LGTM

**Comment by JiafeiPan** [comment]
Updated board document, please review again, thanks.

**Review by carlocaione — DISMISSED** [review]

**Review by povergoing — DISMISSED** [review]

**Comment by JiafeiPan** [comment]
Dear maintainers, update this PR with the following updates, thanks for your review:
Unify SoC name from imx8m/n/p to mimx8m/n/p with the name in hal_nxp.
Add me to be codeowner of platform imx8mm/p/n ARM64 platforms.

**Review by carlocaione — DISMISSED** [review]

**Review by povergoing — DISMISSED** [review]

**Comment by JiafeiPan** [comment]
As hal_nxp patches have been merged, so update west.yml to use actual commit ID.

**Review by carlocaione — DISMISSED** [review]

**Comment by JiafeiPan** [comment]
Dear maintainers, rebased, no dependency now, please review and merge it, thanks.

**Review by carlocaione — APPROVED** [review]

**Review by povergoing — APPROVED** [review]

**Review by SgrrZhf — APPROVED** [review]

**Pull request merged**